app/tests/spec/views/settings/avatar_change.js

```diff
@@ -154,18 +154,16 @@ function (chai, _, $, sinon, View, RouterMock, FileReaderMock, ProfileMock,
           view.afterVisible()
             .then(function () {
               var ev = FileReaderMock._mockBadPngEvent();
-
-              view.router.on('navigate', function () {
-                try {
-                  assert.equal(routerMock.page, 'settings');
-                  assert.equal(view.ephemeralMessages.get('error'), AuthErrors.toMessage('UNUSABLE_IMAGE'));
-                  done();
-                } catch (e) {
-                  return done(e);
-                }
-              });
-
-              view.fileSet(ev);
+              view.fileSet(ev)
+                .fail(function () {
+                  try {
+                    assert.equal(view.$('.error').text(), AuthErrors.toMessage('UNUSABLE_IMAGE'));
+                    assert.isTrue(view.isErrorVisible());
+                    done();
+                  } catch (e) {
+                    return done(e);
+                  }
+                });
```

Comment:
Now that `view.fileSet` is a promise you can remove the `done` callback and simply return `view.fileSet` here and also return `view.afterVisible` so the test will catch any errors.